libs/blocks/merch-cards/merch-cards.js

```diff
@@ -0,0 +1,257 @@
+
+const fail = (el, err = '') => {
+  window.lana?.log(`Failed to initialize merch cards: ${err}`);
+  el.innerHTML = '';
+  return el;
+};
+
+/**
+ * Removes merch cards from the DOM if they are not meant to be displayed in this merch cards block.
+ * @param {*} merchCards merch-cards element
+ */
+export function filterMerchCards(merchCards, filtered) {
+  [...merchCards.children].filter((child) => child.tagName === 'MERCH-CARD')
+    .forEach((card) => {
+      if (!Object.prototype.hasOwnProperty.call(card.filters, filtered)) {
+        card.remove();
+      }
+    });
+}
+
+/** parse card preferences for each category filter */
+export function parsePreferences(elements) {
+  return [...elements].map((el) => {
+    let size;
+    if (el?.firstElementChild?.tagName === 'STRONG') {
+      size = 'wide';
+      if (el.firstElementChild.firstElementChild?.tagName === 'U') {
+        size = 'super-wide';
+      }
+    }
+    return [el.textContent?.trim(), size];
+  });
+}
+
+async function initMerchCards(config, type, filtered, el, preferences) {
+  let cardsData;
+  let err;
+
+  try {
+    const res = await fetch(`${config?.locale?.prefix ?? ''}${config.queryIndexCardPath}.json?sheet=${type}`);
+    if (res.ok) {
+      cardsData = await res.json();
+    } else {
+      err = res.statusText || res.status;
+    }
+  } catch (error) {
+    err = error.message;
+  }
+  if (!cardsData) {
+    fail(el, err);
+  }
+
+  // TODO add aditional parameters.
+  const cards = `<div>${cardsData.data.map(({ cardContent }) => cardContent).join('\n')}</div>`;
+  const fragment = document.createRange().createContextualFragment(cards);
+  const cardsRoot = fragment.firstElementChild;
+  await makePause();
+  // Replace placeholders
+  cardsRoot.innerHTML = await replaceText(cardsRoot.innerHTML, config);
+  await makePause();
+  const autoBlocks = await decorateLinks(cardsRoot).map(loadBlock);
+  await Promise.all(autoBlocks);
+  await makePause();
+  const blocks = [...cardsRoot.querySelectorAll(':scope > div')].map(loadBlock);
+
+  // process merch card blocks in batches of 3.
+  const batchSize = 3;
+  for (let i = 0; i < blocks.length; i += batchSize) {
+    const batch = blocks.slice(i, i + batchSize);
+    await Promise.all(batch);
+    await makePause();
+  }
+
+  if (filtered) {
+    filterMerchCards(cardsRoot, filtered);
+  }
+  // re-order cards, update card filters
+  [...cardsRoot.children].filter((card) => card.tagName === 'MERCH-CARD').forEach((merchCard) => {
+    const filters = { ...merchCard.filters };
+    Object.keys(filters).forEach((key) => {
+      const preference = preferences[key];
+      if (!preference) return;
+      preference
+        .forEach(([cardTitle, cardSize], index) => {
+          if (merchCard.title === cardTitle) {
+            filters[key] = { order: index, size: cardSize };
+          }
+        });
+    });
+    merchCard.filters = filters;
+  });
+  return cardsRoot;
+}
+
+export default async function main(el) {
```

Comment:
Just curious, any reason why `main()` instead of `init`?

Reply:
@Axelcureno, just oversight, changing it 👍